src/codec/xz2/encoder.rs

```diff
@@ -9,6 +9,8 @@ use crate::{
 
 pub struct Xz2Encoder {
     stream: Stream,
+    #[cfg(feature = "xz-parallel")]
+    threads: Option<u32>,
```

Comment:
```suggestion
    threads: Option<NonZeroU32>,
```

Reply:
It seeems like this field isn't used anywhere?

Reply:
It's used in flush. However, a bool would also suffice. The thread count is not actually being used.